apps/meteor/app/apps/server/bridges/rooms.ts

```diff
@@ -102,6 +102,58 @@ export class AppRoomBridge extends RoomBridge {
 		return this.orch.getConverters()?.get('users').convertById(room.u._id);
 	}
 
+	protected async getMessages(
+		roomId: string,
+		options: {
+			limit: number;
+			skip?: number;
+			sort?: Record<string, 1 | -1>;
+		},
+		appId: string,
+	): Promise<IMessage[]> {
+		this.orch.debugLog(`The App ${appId} is getting the messages of the room: "${roomId}"`);
+
+		let { limit, skip = 0, sort } = options;
+
+		if (!Number.isFinite(limit) || limit < 1) {
+			limit = 100;
+		}
+
+		if (!Number.isFinite(skip) || skip < 0) {
+			skip = 0;
+		}
+
+		if (!sort || typeof sort !== 'object') {
+			sort = { ts: -1 };
+		}
+
+		const messageConverter = this.orch.getConverters()?.get('messages');
+		if (!messageConverter) {
+			throw new Error('Message converter not found');
+		}
+
+		const messageQueryOptions = {
+			limit: Math.min(limit, 100),
+			skip,
+			sort,
+		};
+
+		const query = {
+			rid: roomId,
+			_hidden: { $ne: true },
+			t: { $exists: false },
+		};
+
+		const cursor = Messages.find(query, messageQueryOptions);
+
+		const messagePromises: Promise<IMessage>[] = [];
+		await cursor.forEach((message) => {
+			messagePromises.push(messageConverter.convertMessage(message));
```

Comment:
I think we will need to rethink how we fetch the messages here. The message converter itself does at least 2 more queries to the DB - 3 if the message has been edited - which can hit pretty hard if the workspace is very active.

if we have 100 messages returned by mongo, we're looking at an average of 400 other requests sent via the network :grimacing: 

@Dnouv do you have any ideas how we could circumvent this?

The one thing I can think of is to simply passthrough the queried fields (`room`, `sender`, `editor`) without querying for the full data, leaving it to the app developer to do the queries if they need to.

The message converter, for reference https://github.com/RocketChat/Rocket.Chat/blob/33cd125419f254f5093b3de9f68ab13f534e3a28/apps/meteor/app/apps/server/converters/messages.js#L39-L74

Reply:
Added a new converter `convertMessageRaw` it removes a few fields from the following:
```
sender: {
  id: string,
  username: string,
  name?: string
},
editor: {
  id: string,
  username: string
},
room: {
  id: string
}
```

Thank you!